util/src/main/java/org/killbill/billing/util/entity/dao/DefaultPaginationSqlDaoHelper.java

```diff
@@ -85,15 +106,16 @@ public Long inTransaction(final EntitySqlDaoWrapperFactory entitySqlDaoWrapperFa
             // The count to get maxNbRecords can be expensive on very large datasets. As a heuristic to check how large that number is,
             // we retrieve 1 record at offset simplePaginationThreshold (pretty fast). If we've found a record, that means the count is larger
             // than this threshold and we don't issue the full count query
-            final Long recordIdAtSimplePaginationOffset = sqlDao.getRecordIdAtOffset(simplePaginationThreshold);
+            final Long recordIdAtSimplePaginationOffset = withAccountRecordId ? sqlDao.getRecordIdAtOffsetWithAccountRecordId(simplePaginationThreshold, context) : sqlDao.getRecordIdAtOffset(simplePaginationThreshold);
```

Comment:
Could you add a few more test cases in `TestDefaultPaginationSqlDaoHelper`? I think it's worth testing it (e.g. insert 10 records, and go through them 2 at a time to verify the `Pagination` fields).

Reply:
I agree, spending time writing some good DAO tests to verify all these queries is a good investment.

Reply:
I have now added some tests to `TestDefaultPaginationSqlDaoHelper`. Please take a look and let me know how these look. Also, there are some tests in [TestInvoiceDao](https://github.com/reshmabidikar/killbill/blob/ac2e822b14844eea8f88cc5b8e775020975efa53/invoice/src/test/java/org/killbill/billing/invoice/dao/TestInvoiceDao.java#L248) as well.